integration/simulation/simulation_test.go

```diff
@@ -1,306 +1,63 @@
 package simulation
 
 import (
-	"fmt"
 	"math/rand"
-	"os"
 	"testing"
 	"time"
 
-	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
-	"github.com/obscuronet/obscuro-playground/go/obscuronode/host"
-	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
+	"github.com/google/uuid"
+)
+
+// This test creates a network of in memory L1 and L2 nodes, then injects transactions, and finally checks the resulting output blockchain.
+// Running it long enough with various parameters will test many corner cases without having to explicitly write individual test for them.
+// The unit of time is the "avgBlockDurationUSecs" - which is the average time between L1 blocks. Everything else is reported to this value.
+// This number has to be adjusted in conjunction with the number of nodes. If it's too low, the CPU usage will be very high during the simulation
+// which might result in inconclusive results.
+func TestInMemoryMonteCarloSimulation(t *testing.T) {
+	// define core test parameters
+	numberOfNodes := 10
+	numberOfWallets := 5
 
-	"github.com/ethereum/go-ethereum/common"
+	simulationTimeSecs := 15 // in seconds
 
-	"github.com/obscuronet/obscuro-playground/go/log"
+	// This is a critical parameter of the simulation. The value should be as low as possible, as long as the test is still meaningful
+	avgBlockDurationUSecs := uint64(20_000) // in u seconds 1 sec = 1e6 usecs.
 
-	"github.com/google/uuid"
-	"golang.org/x/sync/errgroup"
+	avgNetworkLatency := avgBlockDurationUSecs / 15 // artificial latency injected between sending and receiving messages on the mock network
+	avgGossipPeriod := avgBlockDurationUSecs / 3    // POBI protocol setting
 
-	ethereum_mock "github.com/obscuronet/obscuro-playground/integration/ethereummock"
-	wallet_mock "github.com/obscuronet/obscuro-playground/integration/walletmock"
-)
+	// converted to Us
+	simulationTimeUSecs := simulationTimeSecs * 1000 * 1000
 
-func TestSimulation(t *testing.T) {
 	rand.Seed(time.Now().UnixNano())
 	uuid.EnableRandPool()
 
 	logFile := setupTestLog("../.build/simulations/")
 	defer logFile.Close()
 
-	// define core test parameters
-	numberOfNodes := 10
-	simulationTimeSecs := 15 // in seconds
-	// This is 25 times the measured time (~6 millis) for sending a rollup over the network when running the simulation.
-	avgBlockDurationUSecs := uint64(150_000) // in u seconds 1 sec = 1e6 usecs.
-	avgLatency := avgBlockDurationUSecs / 15
-	avgGossipPeriod := avgBlockDurationUSecs / 3
+	stats := NewStats(numberOfNodes) // todo - temporary object used to collect metrics. Needs to be replaced with something better
```

Comment:
Can you explain why this needs "something better"? Doesn't seem to be explained what's wrong with it anywhere.

Reply:
Added the explanation where the Stats type is defined